### PR TITLE
Fix: Need help url on Grid Container after refresh [ED-13031]

### PR DESCRIPTION
**Pull request by ronenboxer**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: tests/playwright/sanity/container-grid.test.ts [hunk 1]
```diff
@@ -700,6 +700,15 @@ test.describe( 'Container Grid tests @container', () => {
 			expect.soft( linkElement ).toHaveAttribute( 'href', 'https://go.elementor.com/widget-container-grid' );
 		} );
 
+		await test.step( 'Select existing grid container', async () => {
```
Comment (davseve) [inline tests/playwright/sanity/container-grid.test.ts:703]: Do we want to check what happens if we change from flex to grid?

Comment (ronenelementor) [reply tests/playwright/sanity/container-grid.test.ts:703]: It is tested on line 672.
This test is to make sure that the link is rendered correctly after selecting other elements

**Review by davseve — DISMISSED** [review]

File: assets/dev/js/editor/elements/views/container.js [hunk 2]
```diff
@@ -57,6 +57,8 @@ const ContainerView = BaseElementView.extend( {
 			directionSettingKey = this.getDirectionSettingKey() + deviceSuffix,
 			currentDirection = this.container.settings.get( directionSettingKey );
 
+		this.updateNeedHelpLink();
```
Comment (louiswol94) [inline assets/dev/js/editor/elements/views/container.js:60]: Hey @ronenelementor Nice investigation, but this is unfortunately not the correct place to add this code.

There is a JS hook that runs when the panel opens: `panel/open_editor/container`, you can listen to this hook and update the link from there. This hook and callback can be added in `grid-container.js`

Then you can also update `views/container.js` and remove `this.updateNeedHelpLink()` in the `onRender()` method, it is not needed as far as I can see.

So now we covered all scenarios:
1) When the panel opens
2) When the Container Type setting is changed.

I will attach screenshots of the proposed diff, but please test it thoroughly afterwards to make sure everything is covered in the mind map (DnD, refresh, setting change, import etc).

Screenshot 2023-12-11 at 10 31 46

Screenshot 2023-12-11 at 10 32 15

**Review by louiswol94 — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/elements/views/container.js (hunk 2 above)
Comment (ronenelementor) [reply assets/dev/js/editor/elements/views/container.js:60]: Will do! Thanks!

**Review by louiswol94 — APPROVED** [review]

**Pull request merged**